### PR TITLE
Clean up MacOS Warnings

### PR DESCRIPTION
**Pull request by wesbland**
Clean up a bunch of warnings when using more recent versions of GCC and Clang on MacOS.

### CODE REVIEW

**Comment by wesbland** [comment]
test:jenkins/ch4/most
src/mpid/ch4/src/ch4_coll.h

**Comment by wesbland** [comment]
test:jenkins/ch3/most

File: src/mpid/ch4/netmod/portals4/globals.c [hunk 1]
```diff
@@ -11,4 +11,5 @@
 #include "ptl_impl.h"
 #include "ptl_types.h"
 
-MPIDI_PTL_global_t MPIDI_PTL_global = { 0 };
+MPIDI_PTL_global_t MPIDI_PTL_global = { {0}
+};
```
Comment (pavanbalaji) [inline src/mpid/ch4/netmod/portals4/globals.c]: You might want to move the curly brace up and add indent off here:

```
/* *INDENT-OFF* */
MPIDI_POSIX_eager_fbox_control_t MPIDI_POSIX_eager_fbox_control_global = { {0} };
/* *INDENT-ON* */
```

**Comment by pavanbalaji** [comment]
@wesbland: I have some more warning squashes locally.  Can I push them to the same branch?

**Comment by wesbland** [comment]
> @wesbland: I have some more warning squashes locally. Can I push them to the same branch?

Go for it. Feel free to move the curly brace as mentioned above if you like. 😄

**Comment by wesbland** [comment]
@pavanbalaji I'm seeing lots like this too, but fixing them is pretty tricky and I didn't have time for this PR:

```
CC       src/mpi/init/lib_libmpi_la-abort.lo
In file included from /usr/local/Cellar/gcc/8.2.0/lib/gcc/8/gcc/x86_64-apple-darwin18.2.0/8.2.0/include-fixed/stdio.h:425,
                 from ./src/include/mpiimpl.h:22,
                 from src/mpi/init/abort.c:8:
src/mpi/init/abort.c: In function 'MPI_Abort':
src/mpi/init/abort.c:135:38: warning: '%s' directive output may be truncated writing up to 127 bytes into a region of size 71 [-Wformat-truncation=]
         MPL_snprintf(abort_str, 100, "application called MPI_Abort(%s, %d) - process %d", comm_name,
                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~
src/mpi/init/abort.c:135:9: note: in expansion of macro 'MPL_snprintf'
         MPL_snprintf(abort_str, 100, "application called MPI_Abort(%s, %d) - process %d", comm_name,
         ^~~~~~~~~~~~
/Users/wbland/code/mpich/src/mpl/include/mpl_str.h:23:22: note: '__builtin_snprintf' output between 46 and 193 bytes into a destination of size 100
 #define MPL_snprintf snprintf
                      ^~~~~~~~
src/mpi/init/abort.c:135:9: note: in expansion of macro 'MPL_snprintf'
         MPL_snprintf(abort_str, 100, "application called MPI_Abort(%s, %d) - process %d", comm_name,
         ^~~~~~~~~~~~
```

**Comment by wesbland** [comment]
There's also new warnings from https://github.com/pmodels/mpich/pull/3529

**Comment by pavanbalaji** [comment]
> There's also new warnings from #3529

Yup, I got those in my fixes too.  Should I force push?

**Comment by pavanbalaji** [comment]
> @pavanbalaji I'm seeing lots like this too, but fixing them is pretty tricky and I didn't have time for this PR:
> 
> ```
> CC       src/mpi/init/lib_libmpi_la-abort.lo
> In file included from /usr/local/Cellar/gcc/8.2.0/lib/gcc/8/gcc/x86_64-apple-darwin18.2.0/8.2.0/include-fixed/stdio.h:425,
>                  from ./src/include/mpiimpl.h:22,
>                  from src/mpi/init/abort.c:8:
> src/mpi/init/abort.c: In function 'MPI_Abort':
> src/mpi/init/abort.c:135:38: warning: '%s' directive output may be truncated writing up to 127 bytes into a region of size 71 [-Wformat-truncation=]
>          MPL_snprintf(abort_str, 100, "application called MPI_Abort(%s, %d) - process %d", comm_name,
>                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~
> src/mpi/init/abort.c:135:9: note: in expansion of macro 'MPL_snprintf'
>          MPL_snprintf(abort_str, 100, "application called MPI_Abort(%s, %d) - process %d", comm_name,
>          ^~~~~~~~~~~~
> /Users/wbland/code/mpich/src/mpl/include/mpl_str.h:23:22: note: '__builtin_snprintf' output between 46 and 193 bytes into a destination of size 100
>  #define MPL_snprintf snprintf
>                       ^~~~~~~~
> src/mpi/init/abort.c:135:9: note: in expansion of macro 'MPL_snprintf'
>          MPL_snprintf(abort_str, 100, "application called MPI_Abort(%s, %d) - process %d", comm_name,
>          ^~~~~~~~~~~~
> ```

I'm not seeing these.  In fact, I wasn't seeing any warnings from many of your fixes either.  With my set of changes, I was fully warning free.

**Comment by pavanbalaji** [comment]
Force pushed some more warning squashes.  Restarting tests.

test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by pavanbalaji** [comment]
Tests looks good.  @wesbland, can you look at the patches that I added?  After that this will be ready to merge.

**Comment by wesbland** [comment]
Which compiler/version/flags are you using?

I'm using GCC 8.2.0 with `--enable-strict`.

```
$ gcc-8 --version
gcc-8 (Homebrew GCC 8.2.0) 8.2.0
```

```
$ ./configure --prefix=/Users/wbland/code/mpich-ofi/_inst --enable-shared --enable-static --enable-error-messages=yes --enable-large-tests --enable-strict --enable-collalgo-tests --with-zm-prefix=yes --enable-g=all --enable-error-checking=all --enable-fast=none --with-device=ch4:ofi --enable-ch4-netmod-inline=no --enable-ch4-shm-inline=no --with-libfabric=embedded CC=gcc CXX=g++ FC=gfortran F77=gfortran
```

**Comment by wesbland** [comment]
I have no idea if the tests are passing. Can we please add all of these RMA tests to the xfail or merge  #3493 or #3541 this morning?

**Comment by wesbland** [comment]
I can't approve my own PRs, but this looks good to me. If the tests look ok to someone who knows what's going on, this can be merged.

**Comment by pavanbalaji** [comment]
> I have no idea if the tests are passing. Can we please add all of these RMA tests to the xfail or merge #3493 or #3541 this morning?

These really need to be merged.  The RMA tests are getting annoying and are stalling a lot of other work.  @yfguo are those close to getting merged today?

**Comment by pavanbalaji** [comment]
> I can't approve my own PRs, but this looks good to me. If the tests look ok to someone who knows what's going on, this can be merged.

The tests look fine.  Can I merge what we have first so I'm not blocking on them?

**Comment by wesbland** [comment]
Go for it.

**Pull request merged**